plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -1,19 +1,258 @@
 package com.google.cloud.tools.eclipse.appengine.deploy.ui;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Text;
 import org.eclipse.ui.IWorkbenchPropertyPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
 import org.eclipse.ui.dialogs.PropertyPage;
+import org.eclipse.ui.forms.widgets.ExpandableComposite;
+
+import com.google.cloud.tools.eclipse.ui.util.FontUtil;
 
 public class DeployPropertyPage extends PropertyPage implements IWorkbenchPropertyPage {
 
+  private static final String APPENGINE_DASHBOARD_URL = "https://console.developers.google.com/appengine";
+  private static final int INDENT_CHECKBOX_ENABLED_WIDGET = 10;
+
+  private Button promptForProjectIdButton;
+  private Label projectIdLabel;
+  private Text projectId;
+  private Button browseProjectButton;
+  
+  private Button overrideDefaultVersionButton;
+  private Label versionLabel;
+  private Text version;
+  
+  private Button autoPromoteButton;
+  
+  private Button overrideDefaultBucketButton;
+  private Label bucketLabel;
+  private Text bucket;
+  private Button browseBucketButton;
+  
   @Override
   protected Control createContents(Composite parent) {
-    Label label = new Label(parent, SWT.LEFT);
-    label.setText(Messages.getString("preferences.title")); //$NON-NLS-1$
-    return label;
+    Composite container = new Composite(parent, SWT.NONE);
+    container.setLayout(new GridLayout(1, false));
+
+    createProjectIdSection(container);
+
+    createProjectVersionSection(container);
+
+    createPromoteSection(container);
+
+    createAdvancedSection(container);
+    
```

Comment:
`#applyDialogFont()` is a bit strange: it only applies the font if the dialog font is different from the 'default' font.  Which makes sense, providing all widgets use the default font.  But `Link` doesn't choose the default font.